CHANGES.md

```diff
@@ -0,0 +1,6303 @@
+    use this, see Reference: Function. For technical details, e.g. why
+    we did not choose the syntax `(a,b) -> a+b`, see
+    [#490](https://github.com/gap-system/gap/pull/490).
+
+  - Function calls, list accesses and records accesses now can be
+    nested. For example, you can now write `y := f().x;` (essentially
+    equivalent to `y := f();; y := y.x;`), which previously would have
+    resulted in an error; see
+    [#457](https://github.com/gap-system/gap/issues/457) and
+    [#462](https://github.com/gap-system/gap/pull/462)).
+
+  - The libraries of small, primitive and transitive groups which
+    previously were an integral part of GAP were split into three
+    separate packages
+    [PrimgGrp](http://gap-packages.github.io/primgrp/),
+    [SmallGrp](https://gap-packages.github.io/smallgrp/) and
+    [TransGrp](http://www.math.colostate.edu/~hulpke/transgrp/). For
+    backwards compatibility, these are required packages in GAP 4.9
+    (i.e., GAP will not start without them). We plan to change this
+    for GAP 4.10 (see
+    [#2434](https://github.com/gap-system/gap/pull/2434)), once all
+    packages which currently implicitly rely on these new packages had
+    time to add explicit dependencies on them
+    ([#1650](https://github.com/gap-system/gap/pull/1650),
+    [#1714](https://github.com/gap-system/gap/pull/1714)).
+
+  - The performance of GAP's sorting functions (such as `Sort`
+    (Reference: Sort), `SortParallel` (Reference: SortParallel), etc.)
+    has been substantially improved, in some examples by more than a
+    factor of four: as a trivial example, compare the timing for
+    `Sort([1..100000000] * 0)`. As a side effect, the result of sorting
+    lists with equal entries may produce different answers compared to
+    previous GAP versions. If you would like to make your code
+    independant of the exact employed sorting algorithm, you can use the
+    newly added `StableSort` (Reference: StableSort), `StableSortBy`
+    (Reference: StableSortBy) and `StableSortParallel` (Reference:
+    StableSortParallel). (For some technical details, see
+    [#609](https://github.com/gap-system/gap/pull/609)).
+
+  - We removed our old home-grown big integer code, and instead always
+    use the GMP based big integer code. This means that the GMP library
+    now is a required dependency, not just an optional one. Note that
+    GAP has been using GMP big integer arithmetic for a long time by
+    default, and we also have been bundling GMP with GAP. So this change
+    mostly removed code that was never in use for most users.
+
+  - A number of improvements have been made to `Random` (Reference:
+    Random). These may lead to different sequences of numbers being
+    created. On the up side, many more methods for `Random` (Reference:
+    Random) (and other `RandomXYZ` operations) now optionally take an
+    explicit `RandomSource` (Reference: RandomSource) as first argument
+    (but not ye
```

Comment:
Replace by a version - independent link to a file from http://www.math.rwth-aachen.de/~Frank.Luebeck/gap/GAPDoc/doc/chap0.html